### ISSUE TITLE
salt-cloud show_instance action fails on EC2 instances

### ISSUE THREAD

**Issue by sergeizv** [saltstack/salt#39782]
### Description of Issue/Question

'salt-cloud -a show_instance' fails with TypeError on EC2 instances. Releases starting from v2016.3.0 are affected.

It's reborn #33162, which was fixed by #33164 on 2015.8 but appeared again on 2016.3 after #33164 was merged into it (679200aeb2fd) because of changed return value of _get_node.

### Steps to Reproduce Issue

Given an EC2 instance named 'foo1'

```
(py2-env) salt-test@tommynaut:~> salt-cloud -l debug -y -a show_instance foo1
...                                                                          
[ERROR   ] There was an error actioning machines: 'NoneType' object has no attribute '__getitem__'
Traceback (most recent call last):
  File "/home/salt-test/salt/salt/cloud/cli.py", line 252, in run
    ret = mapper.do_action(names, kwargs)
  File "/home/salt-test/salt/salt/cloud/__init__.py", line 1527, in do_action
    vm_name, call='action'
  File "/home/salt-test/salt/salt/cloud/clouds/ec2.py", line 3206, in show_instance
    __opts__)
  File "/home/salt-test/salt/salt/utils/cloud.py", line 2919, in cache_node
    path = os.path.join(prov_dir, '{0}.p'.format(node['name']))
TypeError: 'NoneType' object has no attribute '__getitem__'

```

### Versions Report

```
Salt Version:
           Salt: 2016.3.5-94-g4ee59be
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.5
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.13 (default, Jan 03 2017, 17:41:54) [GCC]
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.1.6
 
System Versions:
           dist:   
        machine: x86_64
        release: 4.9.11-1-default
         system: Linux
        version: Not Installed

```

**Comment by Ch3LL** [comment]
@sergeizv thanks for the PR on this. I'll go ahead and close this issue now. Thanks once again!
